catroid/src/main/java/org/catrobat/catroid/ui/SpriteActivity.java

```diff
@@ -783,6 +789,11 @@ public void onCheckedChanged(CompoundButton compoundButton, boolean checked) {
 	}
 
 	public void handlePlayButton(View view) {
+		if (getCurrentFragment() instanceof ScriptFragment
```

Comment:
as with the other suggestion I think you could unify something here. i.e. move both cases into one if block that checks if the user is in the `ScriptFragment` and then go from there. this would make it easier to read the block.

alternatively you could also just introduce a `boolean` variable for the condition `getCurrentFragment() instanceof ScriptFragment && ((ScriptFragment) getCurrentFragment()).isCurrentlyHighlighted()` so that both conditions stay separate, but look similar. 

you can do whichever you prefer :wink: 

```
if (isScriptFragment) {
    if (areBricksHighlighted) {
        ...
    }
    if (dragAndDropActive) {
        ...
    }
}
```
or 
```
if (areBricksHighlightedInFragment) {
    ...
}
if (dragAndDropActiveInFragment) {
    ...
}
```